deps/rabbit/src/rabbit_amqp_management.erl

```diff
@@ -159,7 +159,7 @@ handle_http_req(HttpMethod = <<"PUT">>,
                     %% Must have been created in the meantime. Loop around again.
                     handle_http_req(HttpMethod, PathSegments, Query, ReqPayload,
                                     Vhost, User, ConnPid, {PermCache2, TopicPermCache});
-                {protocol_error, precondition_failed = _ErrorType, Reason, ReasonArgs} ->
+                {protocol_error, precondition_failed = _ErrorType, Reason, [_QueueName, _VHost, _Limit] = ReasonArgs} ->
```

Comment:
You don't have to destructure `ReasonArgs` if the individual parts are not used.

Reply:
We could rename the returned error `{protocol_error, precondition_failed, Reason, ReasonArgs}` to something like `{error, queue_limit_exceeded, Reason, ReasonArgs}` so that each caller of `rabbit_queue_type:declare/2` can handle this error correctly?
Here, it maps better to status code `403 Forbidden` than to `412 Precondition Failed`.
https://datatracker.ietf.org/doc/html/rfc7232#section-4.2 doesn't make sense here:
>    The 412 (Precondition Failed) status code indicates that one or more
   conditions given in the request header fields evaluated to false when
   tested on the server.  This response code allows the client to place
   preconditions on the current resource state (its current
   representations and metadata) and, thus, prevent the request method
   from being applied if the target resource is in an unexpected state.